File: programmers/난이도별/level01.콜라츠_추측/JiaLee0707.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int num) {
    int answer = 0;
    long long number = num;

    for (answer = 0; answer < 500; answer++) {
        if (number == 1) return answer;
        else if (number % 2 == 0) number /= 2;
        else number = number * 3 + 1;
    }

    return -1;
}

File: programmers/난이도별/level01.콜라츠_추측/JiaLee0707.java
class Solution {
    public int solution(int num) {
        long answer = num;
        for(int i=0; i<500; i++) {
            if(answer == 1) return i;
            answer = answer%2==0 ? answer/2 : answer*3+1;
        }
        
        return -1;
    }
}
